istioctl/pkg/waypoint/waypoint.go

```diff
@@ -480,13 +481,31 @@ func deleteWaypoints(cmd *cobra.Command, kubeClient kube.CLIClient, namespace st
 			if gw.Spec.GatewayClassName != constants.WaypointGatewayClassName {
 				continue
 			}
-			names = append(names, gw.Name)
+			nameList = append(nameList, gw.Name)
+		}
+	} else {
+		gws, err := kubeClient.GatewayAPI().GatewayV1().Gateways(namespace).List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+
+		for _, name := range names {
+			contain, gw := waypointContains(gws.Items, name)
+			if contain {
+				if gw.Spec.GatewayClassName != constants.WaypointGatewayClassName {
+					fmt.Fprintf(cmd.OutOrStdout(), "waypoint %v/%v not found\n", namespace, name)
```

Comment:
This log can be more accurate, like indicating it's not waypoint.

Reply:
Yes, but this conflicts with the previous comment. After building a valid map, it is difficult to distinguish whether there is no gateway or the gateway class does not match.